### PR TITLE
Common ECC608A folder

### PR DESCRIPTION
**Pull request by lundinc2**
Description
-----------

Checklist:
----------

- [ ] I have tested my changes. No regression in existing tests.
- [ ] My code is Linted.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/abstractions/pkcs11/ecc608a/iot_pkcs11_secure_element.c [hunk 1]
```diff
@@ -220,6 +216,17 @@ ATCAIfaceCfg cfg_ateccx08a_i2c_default = {
     .rx_retries             = 20
 };
 
+/** \brief default configuration for Kit protocol over the device's async interface */
+ATCAIfaceCfg cfg_ateccx08a_kithid_default = {
+    .iface_type         = ATCA_HID_IFACE,
+    .devtype            = ATECC508A,
```
Comment (dcgaws) [inline libraries/abstractions/pkcs11/ecc608a/iot_pkcs11_secure_element.c]: Why do we need this?

Comment (lundinc2) [reply libraries/abstractions/pkcs11/ecc608a/iot_pkcs11_secure_element.c]: This is the ecc608 configuration used the windows port. I made a copy paste error here, and will update with the correct HID configuration.

Comment (lundinc2) [reply libraries/abstractions/pkcs11/ecc608a/iot_pkcs11_secure_element.c]: Also I think we should add a flag to make sure we don't waste memory compiling the other port's ecc608a configuration.

Comment (lundinc2) [reply libraries/abstractions/pkcs11/ecc608a/iot_pkcs11_secure_element.c]: @dcgaws I fixed the copy paste error, and added macros to only compile the correct ecc608a config.

**Pull request merged**